README.md

```diff
@@ -1,2 +1,177 @@
+
+## Future work
+
+### Async parsing
+
+It's often better to let the browser parse resources in parallel with other work if possible. We can adjust the transform to use `CSSStyleSheet.prototype.replace()` instead of `replaceSync()` so that the CSS parsing happens off the main thread.
+
+This is possible with top level await, so that the transformed module would look like:
+
+`styles.css`:
+```js
+const stylesheet = new CSSStyleSheet();
+await stylesheet.replace(`.foo {\n  color: red;\n}\n`);
+export default stylesheet;
+```
+
+First, the performance impact should be measured. The top-level await will block other JavaScript execution in the same module graph, so it may not help that much. On the other hand, other main thread work can resume. This would be an opt-in feature.
+
+### Dynamic import support
+
+Dynamic imports can be generically supported by transforming them into a `fetch()` call:
+
+`load-css.js`:
+```ts
+const loadCSS = (url) => import(url);
+```
+
+Can be tranformed to:
+`load-css.js`:
+```ts
+const loadCSS = (url) => () => {
+  // A cache of URL -> Promise<CSSSyleSheet>
+  const cache = globalThis._$CSSModuleCache ??= new Map();
+  const resolvedURL = new URL(url, import.meta.url).href;
+  let stylesheet = cache.get(resolvedURL);
+  if (stylesheet !== undefined) {
+    return stylesheet;
+  }
+  stylesheet = new CSSStyleSheet();
+  const promise = (async () => {
+    const response = await fetch(resolvedURL);
+    const text = await response.text();
+    await stylesheet.replace(text);
+    return stylesheet;
+  })();
+  cache.set(resolvedURL, promise);
+  return promise;
+};
+```
+_This is completely untested code_
+
+### External CSS support
+
+Specifiers marked as external in the Rollup config are correctly left unmodified, but by transforming the CSS module to use `fetch()` and top-level await, we can load external CSS files.
+
+### Bundling CSS into CSS
+
+We can leave CSS external to the JavaScript bundle, in order to take enable the browser to directly parse the CSS, but still bundle all the CSS together while letting each import site receive just the `CSSStyleSheet` object it would have pre-transformation by using [this hack](https://github.com/w3c/csswg-drafts/issues/5629#issuecomment-1401322544) to bundle multiple CSS sheets into one file.
+
+### PostCSS integration
+
+Existing PostCSS Rollup plugins may rely on stylesheets being added to the global document via a `<link>` tag. We could add a PostCSS pass to this plugin so that developers could write CSS that's not supported by browsers natively, or run transforms like Tailwind.
+
+We need to be careful that we don't encourage packages to publish non-standard CSS that requires the use of the transform, however. PostCSS transformation may belong in a separate plugin.
+
+### Transform import attributes to import assertions
+
+To support Chrome version before they land import attribute support, we could transform just the import assertion syntax into import attributes.
```

Comment:
my thoughts are it's probably better to just let the old syntax die so not even worth considering this, but i suppose no harm in just listing it among potential future work.

Reply:
Good point there. It would mess w/ metrics allowing removal of the syntax. I hope they add attribute support sooner rather than later.